api/envoy/service/auth/v3/external_auth.proto

```diff
@@ -103,6 +103,15 @@ message OkHttpResponse {
   // to the downstream client on success. Note that the :ref:`append field in HeaderValueOption <envoy_v3_api_field_config.core.v3.HeaderValueOption.append>`
   // defaults to false when used in this message.
   repeated config.core.v3.HeaderValueOption response_headers_to_add = 6;
+
+  // This field allows the authorization service to set (and overwrite) query
+  // string parameters on the original request before it is sent upstream.
+  repeated config.core.v3.QueryParameter query_parameters_to_set = 7;
+
+  // This field allows the authorization service to specify which query parameters
+  // should be removed from the original request before it is sent upstream. Each
+  // element in this list is a case-sensitive query parameter name to be removed.
```

Comment:
Presumably `query_parameters_to_set` is also case sensitive? LGTM API other than this nit.

Reply:
I think the right thing to do is make sure that config.core.v3.QueryParameter makes it clear that `key` is case sensitive.